### PR TITLE
ARC-1059 url params revert and JWT token fix

### PR DESCRIPTION
**Pull request by mboudreau**
Fixed the JWT token generation which wasn't using the correct path/query to generate the token from, causing a 401 http status from the jira API.

As part of this changed, I've also changed the url variable syntax to be more specific by using `{var}` to prevent some edge cases with replacements.

### CODE REVIEW

**Comment by KAbakumovAtl** [comment]
Could you point to the places in the code which were causing issues and how it was fixed?

Also, can we add tests for it?

File: src/jira/client/axios.ts [hunk 1]
```diff
@@ -1,74 +1,60 @@
...
-					Authorization: `JWT ${jwtToken}`
-				}
-			};
-		}
-	);
-}
+const getAuthMiddleware = (secret: string, instance: AxiosInstance) =>
+	(config: AxiosRequestConfig): AxiosRequestConfig => {
+		// Generate full URI based on current config
+		const uri = instance.getUri(config);
```
Comment (mboudreau) [inline src/jira/client/axios.ts:22]: @KAbakumovAtl This is specifically where the problem lied.  It was using `config.url` to parse the path and querystring, but that's an optional parameter and most would use the `params` option instead for queries instead of adding it inline.  By doing this, the path/query could be wrong depending on the situation.

Now we can actually get the full URI that will be generated by Axios before the request and parse that instead since there is an encoding step that happens for all params.

As for tests, there's probably a few we can add around the axios client since we'd need to test it in conjunction with the other middleware which can interfere.  I'll add some in.

Comment (KAbakumovAtl) [reply src/jira/client/axios.ts]: That sounds like it could affect Push API as well. Is it because we didn't use query parameters and the url didnt' change?

File: src/github/client/github-client.ts [hunk 2]
```diff
@@ -132,7 +132,7 @@ export default class GitHubClient {
 	 * Lists pull requests for the given repository.
 	 */
 	public async getPullRequests(owner: string, repo: string, pullRequestParams: GetPullRequestParams): Promise<AxiosResponse<Octokit.PullsListResponseItem[]>> {
-		return await this.get<Octokit.PullsListResponseItem[]>(`/repos/:owner/:repo/pulls`, pullRequestParams, {
+		return await this.get<Octokit.PullsListResponseItem[]>(`/repos/{owner}/{repo}/pulls`, pullRequestParams, {
```
Comment (KAbakumovAtl) [inline src/github/client/github-client.ts:135]: This is GitHub Client. How would this format change impact requests?

File: src/jira/client/axios.ts (hunk 1 above)
Comment (mboudreau) [reply src/jira/client/axios.ts]: yep.

File: test/unit/util/axios/url-params-middleware.test.ts [hunk 3]
```diff
@@ -0,0 +1,110 @@
+import { urlParamsMiddleware } from "../../../../src/util/axios/url-params-middleware";
+import { AxiosRequestConfig } from "axios";
+
+describe("URL Params Axios Middleware", () => {
```
Comment (KAbakumovAtl) [inline test/unit/util/axios/url-params-middleware.test.ts:4]: Nice

**Review by yvonnevictoria — APPROVED** [review]

File: test/unit/util/axios/jira-auth-middleware.test.ts [hunk 4]
```diff
@@ -0,0 +1,69 @@
+import axios, { AxiosRequestConfig } from "axios";
+import fs from "fs";
+import { jiraAuthMiddleware } from "../../../../src/util/axios/jira-auth-middleware";
+import path from "path";
+
+describe("Jira Auth Axios Middleware", () => {
```
Comment (KAbakumovAtl) [inline test/unit/util/axios/jira-auth-middleware.test.ts:6]: Nice

**Review by KAbakumovAtl — APPROVED** [review]
Looks good.

I'd only ask to test changes in staging before merging

File: src/github/client/github-client.ts (hunk 2 above)
Comment (mboudreau) [reply src/github/client/github-client.ts:135]: It won't.  Github client uses Axios with URL Param middleware, so it'll do the correct replacements.
